client/components/course/Sidebar/Relationship.vue

```diff
@@ -41,14 +41,20 @@ export default {
     allowEmpty: { type: Boolean, default: true },
     placeholder: { type: String, default: 'Click to select' },
     allowCircularLinks: { type: Boolean, default: false },
-    allowInsideLineage: { type: Boolean, default: false }
+    allowInsideLineage: { type: Boolean, default: false },
+    allowedTypes: { type: Array, default: () => ([]) }
```

Comment:
Semantically, an empty array would mean no types are allowed and `null` would mean rule not defined. That being said no types allowed would make no sense here so we could let it slide for this one.
Actually, let me tag @abasic for this to weigh in.